pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -17,20 +17,173 @@
+}
+
+// StoragePoolSpec is the spec for a StoragePool resource
+type StoragePoolSpec struct {
+	// StorageCohortReference points to the StorageCohort resource, the pool is to be created upon
+	// +required
+	StorageCohortReference corev1.ObjectReference `json:"storageCohortReference"`
+
+	// Type refers to the type of pool we are expecting to be created by the provisioner
+	// +required
+	Type string `json:"type"`
+
+	// Configuration points to the configuration, a custom resource, map of parameters or configmap
+	// that can be used to specify the pool and its device related configuration.
+	// +required
+	Configuration interface{} `json:"configuration"`
+
+	// Capabilities to be supported by the StoragePool
+	// +optional
+	Capabilities Capabilities `json:"capabilities,omitempty"`
+
+	// RequestedCapacity needed for the pool creation
+	// +required
+	RequestedCapacity resource.Quantity `json:"requestedCapacity"`
+}
+
+// StoragePoolStatus is for handling status of StoragePool resource
+type StoragePoolStatus struct {
+	// ReferenceResource points to the pre-existing pool resource or the created pool resource after creation of pool.
+	// +optional
+	ReferenceResource corev1.ObjectReference `json:"referenceResource,omitempty"`
+
+	// Capacity of the pool,viz total, used and available capacity
+	// +optional
+	Capacity StorageCapacity `json:"capacity,omitempty"`
+
+	// IOPs of the pool,viz total, provisioned, used and available IOPs
+	// +optional
+	IOPs StorageIOPs `json:"IOPs,omitempty"`
+
+	// VolumeSizeMaxLimit for maximum volume size allowed
+	// +optional
+	VolumeSizeMaxLimit resource.Quantity `json:"volumeSizeMaxLimit,omitempty"`
+
+	// Conditions enlist the conditions of the pool based on various types.
+	// +optional
+	Conditions []StoragePoolCondition `json:"state,omitempty"`
+}
+
+// StorageCapacity for the pool,viz total, used and available capacity
+type StorageCapacity struct {
+	// Total Capacity of the pool
+	// +optional
+	Total resource.Quantity `json:"total,omitempty"`
+
+	// Used Capacity of the pool
+	// +optional
+	Used resource.Quantity `json:"used,omitempty"`
+
+	// Available Capacity of the pool
+	// +optional
+	Available resource.Quantity `json:"available,omitempty"`
+
+	// Provisioned Capacity of the pool
+	// +optional
+	Provisioned resource.Quantity `json:"provisioned,omitempty"`
+}
+
+// StorageIOPs for the pool,viz total, provisioned, used and available capacity
+type StorageIOPs struct {
+	// Total IOPs of the pool
+	// +optional
+	Total uint64 `json:"total,omitempty"`
+
+	// Available IOPs of the pool
+	// +optional
+	Available uint64 `json:"available,omitempty"`
+
+	// Provisioned IOPs of the pool
+	// +optional
+	Provisioned uint64 `json:"provisioned,omitempty"`
+
+	// Used IOPs of the pool
+	// +optional
+	Used uint64 `json:"used,omitempty"`
 }
 
+// StoragePoolCondition describes the condition of a StoragePool at a certain point based on a specific type
+type StoragePoolCondition struct {
```

Comment:
Is this struct similar to K8s condition struct?

Reply:
Yup, But we have to use a separate struct as this struct are specific to every k8s component( be it node, pod, container...). And we cannot typecast a custom data type into another.

Reply:
Ok